gather_istio.sh

```diff
@@ -0,0 +1,224 @@
+  local pilotName
+  pilotName=$(getPilotName "${controlPlaneNamespace}")
+  local podNamespace="${2}"
+
+  echo
+  echo "Collecting Envoy config for pods in ${podNamespace}, control plane namespace ${controlPlaneNamespace}"
+
+  local pods
+  pods="$(oc get pods -n "${podNamespace}" -o jsonpath='{ .items[*].metadata.name }')"
+  for podName in ${pods}; do
+    if [ -z "$podName" ]; then
+        continue
+    fi
+
+    if oc get pod -o yaml "${podName}" -n "${podNamespace}" | grep -q proxyv2; then
+      echo "Collecting config_dump and stats for pod ${podName}.${podNamespace}"
+
+      local logPath=${BASE_COLLECTION_PATH}/namespaces/${podNamespace}/pods/${podName}
+      mkdir -p "${logPath}"
+
+      oc exec "${pilotName}" -n "${controlPlaneNamespace}" -c discovery -- bash -c "/usr/local/bin/pilot-discovery request GET /debug/config_dump?proxyID=${podName}.${podNamespace}" > "${logPath}/config_dump_istiod.json" 2>&1
+      oc exec -n "${podNamespace}" "${podName}" -c istio-proxy -- /usr/local/bin/pilot-agent request GET config_dump > "${logPath}/config_dump_proxy.json" 2>&1
+      oc exec -n "${podNamespace}" "${podName}" -c istio-proxy -- /usr/local/bin/pilot-agent request GET stats > "${logPath}/proxy_stats" 2>&1
+    fi
+  done
+}
+
+function version() {
+  if [[ -n $OSSM_MUST_GATHER_VERSION ]] ; then
+    echo "${OSSM_MUST_GATHER_VERSION}"
+  else
+    echo "0.0.0-unknown"
+  fi
+}
+
+function inspect() {
+  local resource ns
+  resource=$1
+  ns=$2
+
+  echo
+  if [ -n "$ns" ]; then
+    echo "Inspecting resource ${resource} in namespace ${ns}"
+    oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}" -n "${ns}"
+  else
+    echo "Inspecting resource ${resource}"
+    oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}"
+  fi
+}
+
+function inspectNamespace() {
+  local ns
+  ns=$1
+
+  inspect "ns/$ns"
+  for crd in $crds; do
+    inspect "$crd" "$ns"
+  done
+  inspect net-attach-def,roles,rolebindings "$ns"
+}
+
+function main() {
+  #TODO: respect --since and --since-time: https://github.com/openshift/enhancements/blob/master/enhancements/oc/must-gather.md#must-gather-images
+  local crds controlPlanes
+  echo
+  echo "Executing Istio gather script"
+  echo
+
+  versionFile="${BASE_COLLECTION_PATH}/version"
+  echo "openshift-service-mesh/must-gather"> "$versionFile"
+  version >> "$versionFile"
+
+  # TODO: add new name label to servicemesh-operator3 pod and use that instead of the control-plane label
+  operatorNamespace=$(oc get pods --all-namespaces -l control-plane=servicemesh-operator3 -o jsonpath="{.items[0].metadata.namespace}")
+
+  inspect nodes
+
+  # TODO: does this match everything we need?
+  for r in $(oc get clusterroles,clusterrolebindings -l install.operator.istio.io/owning-resource -oname); do
+    inspect "$r"
+  done
+  for r in $(oc get clusterroles,clusterrolebindings -l 'app in (istiod,istio-reader)' -oname); do
+    inspect "$r"
+  done
```

Comment:
I don't think we use the `owning-resource` label. Or maybe we set it, but just because the upstream operator sets it, too.

We should probably inspect resources using `metadata.ownerReferences`. Every resource that the operator creates has an `ownerReference` to an `IstioRevision`.

Reply:
Does it make sense to start using `owning-resource ` label or to create new one so we can query via labels? Using `metadata.ownerReferences` is possible but labels seem to be more natural way.

Reply:
Not sure about owning-resource. It could cause `istioctl` to interact with our resources. Can you check if this is the case? We can, however, add a different label for this.

Reply:
We can add and use the labels `app.kubernetes.io/managed-by: sail-operator` and `app.kubernetes.io/instance: <istio-revision-name(?)>`

Reply:
I added more comments so we can revisit after TP.1 when there is more time. There is https://issues.redhat.com/browse/OSSM-6804 to finish missing pieces. I think we can merge now to unblock work on downstream build.